lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -748,12 +748,13 @@ def on_tls_conf_set(
         # Get the list of stored secrets for this cert
         current_secrets = self.secrets.get_object(scope, cert_type.val)
 
-        # Store cert/key on disk - must happen after opensearch stop for transport certs renewal
-        self.store_tls_resources(cert_type, current_secrets)
-
         if scope == Scope.UNIT:
             # node http or transport cert
-            self.opensearch_config.set_node_tls_conf(cert_type, current_secrets)
+            self.opensearch_config.set_node_tls_conf(
+                cert_type,
+                self.secrets.get(Scope.APP, "keystore-password-ca"),
+                self.secrets.get(scope, f"keystore-password-{cert_type}"),
```

Comment:
we should group all of these under the same secrets where we store the TLS resources (chain, key etc..), differentiating between the ones that are with unit scope vs the ones with the app scope

Reply:
Good you bring that up, as I wanted to add a question here: The idea behind having it as a separate secret was that maybe we want to treat keystore-passwords similar to user-passwords, meaning that it could make sense to - at some point - add an action to set keystore-passwords. Then it would be easier with this structure.

Do you think that makes sense or should I treat it like the other TLS resources and move it to the corresponding TLS secret?

Reply:
The full background is that all charms only have 1 secret for all passwords - but since we have TLS mandatory in opensearch we have the exception of having 2 secrets, 1 for passwords and 1 for TLS resources.

Since those passwords are exclusively related to TLS - I think it makes sense to treat them the same way we treat TLS resources secrets-wise.

Reply:
done